app/controllers/dashboard/reports_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class Dashboard::ReportsController < Dashboard::BaseController
+  def all_files
+    return deny_request unless current_user.admin?
```

Comment:
I would move all these into a `before_action`

Have a look at `app/controllers/dashboard/collections_controller.rb` for an example of how this is done. You can use `except: [:monthly_user_work_versions]` on the end of the before action to authorize only the admin ones